megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -5358,7 +5358,12 @@ public static ToHitData processDefenderSPAs(ToHitData toHit, Entity ae, Entity t
 
     @Override
     public String toDisplayableString(Client client) {
-        return "attacking " + getTarget(client.getGame()).getDisplayName() + " with " + getEntity(client.getGame()).getEquipment(weaponId).getName();
+        try {
```

Comment:
This will certainly prevent an NPE here. I'm still dubious. Won't it just crash later when it tries to find the target or the weapon? If this is only called from the boardview to display some info and we definitely want to prevent exceptions for display info then I'd put the try block in boardview.

Reply:
As far as I can tell, this NPE path only occurs while we're updating the Chat window and get interrupted by, for instance, the ConcurrentMod exception above.  This method appears to only be called from AccessibilityWindow, so as long as we output some string it should be fine.

That said, I can make it work more like the version in AbstractAttackAction.java:
```
        final Targetable target = getTarget(client.getGame());
        return (target == null) ? "Attacking Null Target with id " + getTargetId()
                : "Attacking " + target.getDisplayName();
```
so at least the target's ID gets printed.